plugins/org.jkiss.dbeaver.net.ssh/src/org/jkiss/dbeaver/model/net/ssh/SSHUtils.java

```diff
@@ -78,4 +89,98 @@ public static boolean isKeyEncrypted(byte[] privKeyValue) {
         return false;
     }
 
+    private static final String PLATFORM_SSH_PREFERENCES_NODE = "org.eclipse.jsch.core"; //$NON-NLS-1$
+    private static final String PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY = IConstants.KEY_SSH2HOME;
+    private static final String DEFAULT_SSH_HOME_DIR_NAME = IConstants.SSH_DEFAULT_HOME;
+    private static final String DEFAULT_SSH_HOME_DIR_NAME_WIN_OLD = IConstants.SSH_OLD_DEFAULT_WIN32_HOME;
+    private static final String KNOWN_SSH_HOSTS_FILE_NAME = "known_hosts";
+
+    @NotNull
+    public static File getKnownSshHostsFileOrDefault() {
+        return  getKnownSshHostsFileImpl(true);
+    }
+
+    @Nullable
+    public static File getKnownSshHostsFileOrNull() {
+        return  getKnownSshHostsFileImpl(false);
+    }
+
+    private static File getKnownSshHostsFileImpl(boolean forceFileObjectOnFail) {
+        String sshHomePathString = Platform.getPreferencesService().getString(PLATFORM_SSH_PREFERENCES_NODE, PLATFORM_SSH_PREFERENCES_SSH2HOME_KEY, null, null);
+        if (CommonUtils.isNotEmpty(sshHomePathString)) {
+            try {
+                return Paths.get(sshHomePathString, KNOWN_SSH_HOSTS_FILE_NAME).toFile();
+            } catch (InvalidPathException e) {
+                log.warn("Failed to resolve SSH known hosts file location at " + sshHomePathString, e);
+                if (forceFileObjectOnFail) {
+                    return new File(sshHomePathString + File.pathSeparator + KNOWN_SSH_HOSTS_FILE_NAME);
+                } else{
+                    return null;
+                }
+            }
+        } else {
+            // seems preference path not set at all, so try to resolve it and preserve
+            return resolveDefaultKnownSshHostsFile(forceFileObjectOnFail, true);
+        }
+    }
+
+    private static boolean isOnWindows() {
```

Comment:
Maybe `RuntimeUtils.isWindows()` can be beneficial?

Reply:
Done